ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/HuggingfaceTextEmbeddingTranslator.java

```diff
@@ -155,11 +188,48 @@ public static final class Builder {
          * @param batchifier true to include token types
          * @return this builder
          */
-        public Builder optBatchifier(Batchifier batchifier) {
+        public HuggingfaceTextEmbeddingTranslator.Builder optBatchifier(Batchifier batchifier) {
             this.batchifier = batchifier;
             return this;
         }
 
+        /**
+         * Sets the normalize for the {@link Translator}.
+         *
+         * @param normalize true to normalize the embeddings
+         * @return this builder
+         */
+        public HuggingfaceTextEmbeddingTranslator.Builder optNormalize(boolean normalize) {
+            this.normalize = normalize;
+            return this;
+        }
+
+        /**
+         * Sets the pooling for the {@link Translator}.
+         *
+         * @param poolingMode the pooling model, one of mean_pool, max_pool and cls
+         * @return this builder
+         */
+        public HuggingfaceTextEmbeddingTranslator.Builder optPoolingMode(String poolingMode) {
+            if (!"mean".equals(poolingMode)
+                    && !"max".equals(poolingMode)
+                    && !"cls".equals(poolingMode)
+                    && !"mean_sqrt_len".equals(poolingMode)
+                    && !"weightedmean".equals(poolingMode)) {
```

Comment:
How about using !EnumSet.allOf(PoolingMethod.class).contains(poolingMode);?

Reply:
This is to keep consistent with DJL code. When DJL team release this translator in future, we will delete this class and use DJL text embedding translator class which doesn't have PoolingMethod enum.